File: Shortest Path/Floyd-Warshall/10159번 - 저울/10159.cpp
//10159번: 저울
#include <iostream>

using namespace std;

int n, m;
int goods[101][101];

void input() {
    cin >> n >> m;
    for(int i = 0, a, b; i < m; i++) {
        cin >> a >> b;
        goods[a][b] = 1;
        goods[b][a] = -1;
    }
}

void solve() {
    for(int k = 1; k <= n; k++) {
        for(int i = 1; i <= n; i++) {
            for(int j = 1; j <= n; j++) {
                if(goods[i][k] == goods[k][j] && goods[i][k] != 0) goods[i][j] = goods[i][k];
            }
        }
    }
    
    for(int i = 1; i <= n; i++) {
        int cnt = n - 1;
        for(int j = 1; j <= n; j++) {
            if(goods[i][j] != 0) cnt--;
        }
        cout << cnt << '\n';
    }
}

int main() {
    ios::sync_with_stdio(0);
    cin.tie(0);
    
    input();
    solve();
    
    return 0;
}

File: Shortest Path/Floyd-Warshall/10159번 - 저울/10159.py
import sys

input = sys.stdin.readline

n = int(input())
m = int(input())
goods = [list(0 for _ in range(n)) for _ in range(n)]
for _ in range(m):
    a, b = map(int, input().split())
    a -= 1
    b -= 1
    goods[a][b] = 1
    goods[b][a] = -1

def solve():
    for k in range(n):
        for i in range(n):
            for j in range(n):
                if goods[i][k] == goods[k][j] and goods[i][k] != 0:
                    goods[i][j] = goods[i][k]
    
    for el in goods:
        cnt = n - 1
        for e in el:
            if e != 0:
                cnt -= 1
        print(cnt)

if __name__ == '__main__':
    solve()